src/pages/LoginPage.jsx

```
@@ -1,29 +1,112 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom"
 import Header from "../shared/header";
 import "../styles/LoginPage.css";
+import { Icon } from 'react-icons-kit';
+import { eyeOff } from 'react-icons-kit/feather/eyeOff';
+import { eye } from 'react-icons-kit/feather/eye';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
 
 function LoginPage() {
-    return(
+
+    const [password, setPassword] = useState("");
+    const [username, setUsername] = useState("");
+    const [type, setType] = useState('password');
+    const [icon, setIcon] = useState(eyeOff);
+    const navigateTo = useNavigate();
+
+    const handlePasswordChange = (e) => {
+        const newPassword = e.target.value;
+        setPassword(newPassword);
+    };
+
+    const handleToggle = () => {
+        if (type === 'password') {
+            setIcon(eye);
+            setType('text')
+        } else {
+            setIcon(eyeOff)
+            setType('password')
+        }
+    }
+
+    const handleLogin = async (e) => {
+        e.preventDefault();
+
+        try {
+            const response = await fetch('http://localhost:8000/api/v1/users/login', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({ username, password }),
+            });
+
+            if (response.ok) {
+                // Items page goes to an error? Does not exist?
+                navigateTo("/Items");
+            } else {
+                // Toast error message not working as expected possibly due to the backend not sending the appropriate error responses based on the login request
+                const errorMessage = await response.text();
+                if (errorMessage.includes('username')) {
+                    toast("Username is incorrect", {});
```

You are off to a great start Ray! 
I would address the comments by Roy as they may be security issues. Instead of checking for invalid username or password in the `else` block, I would just throw a toast message with something like `invalid username or password`.

---

Thank you, I went ahead and made the changes. :)